azurelinuxagent/ga/agent_update_handler.py

```diff
@@ -305,7 +305,7 @@ def run(self, goal_state):
                     # as historically we don't support downgrades below daemon versions. So daemon will not pickup that requested version rather start with
                     # installed latest version again. When that happens agent go into loop of downloading the requested version, exiting and start again with same version.
                     #
-                    raise Exception("Can't process the update as the requested version: {0} is < current daemon version: {1}".format(
+                    raise Exception("as the requested version: {0} is < current daemon version: {1}".format(
```

Comment:
Missing "Unable to update Agent:"?

Reply:
we append this msg when we catch the exception.

Reply:
That is confusing. Myabe instead define an AgentUpdateError exception with a "reason" property and initialize the reason to "Requested version... is < ...."? The format the message where you handle the exception.

Reply:
Or actually, you don't need the "reason" property. The type of the exception would be AgentUpdateError and the message  "Requested version... is < ...."

Reply:
updated